lib/rdoc/parser/prism_ruby.rb

```diff
@@ -0,0 +1,975 @@
+  end
+
+  # Returns a pair of owner module and constant name from a given constant path.
+  # Creates owner module if it does not exist.
+
+  def find_or_create_constant_owner_name(constant_path)
+    const_path, colon, name = constant_path.rpartition('::')
+    if colon.empty? # class Foo
+      [@container, name]
+    elsif const_path.empty? # class ::Foo
+      [@top_level, name]
+    else # `class Foo::Bar` or `class ::Foo::Bar`
+      [find_or_create_module_path(const_path, :module), name]
+    end
+  end
+
+  # Adds a constant
+
+  def add_constant(constant_name, rhs_name, start_line, end_line)
+    comment = consecutive_comment(start_line)
+    handle_consecutive_comment_directive(@container, comment)
+    owner, name = find_or_create_constant_owner_name(constant_name)
+    constant = RDoc::Constant.new(name, rhs_name, comment)
+    constant.store = @store
+    constant.line = start_line
+    record_location(constant)
+    handle_modifier_directive(constant, start_line)
+    handle_modifier_directive(constant, end_line)
+    owner.add_constant(constant)
+    mod =
+      if rhs_name =~ /^::/
+        @store.find_class_or_module(rhs_name)
+      else
+        @container.find_module_named(rhs_name)
+      end
+    if mod && constant.document_self
+      a = @container.add_module_alias(mod, rhs_name, constant, @top_level)
+      a.store = @store
+      a.line = start_line
+      record_location(a)
+    end
+  end
+
+  # Adds module or class
+
+  def add_module_or_class(module_name, start_line, end_line, is_class: false, superclass_name: nil)
+    comment = consecutive_comment(start_line)
+    handle_consecutive_comment_directive(@container, comment)
+    return unless @container.document_children
+
+    owner, name = find_or_create_constant_owner_name(module_name)
+    if is_class
+      mod = owner.classes_hash[name] || owner.add_class(RDoc::NormalClass, name, superclass_name || '::Object')
+
+      # RDoc::NormalClass resolves superclass name despite of the lack of module nesting information.
+      # We need to fix it when RDoc::NormalClass resolved to a wrong constant name
+      if superclass_name
+        superclass_full_path = resolve_constant_path(superclass_name)
+        superclass = @store.find_class_or_module(superclass_full_path) if superclass_full_path
+        superclass_full_path ||= superclass_name
+        if superclass
+          mod.superclass = superclass
+        elsif mod.superclass.is_a?(String) && mod.superclass != superclass_full_path
+          mod.superclass = superclass_full_path
+        end
+      end
+    else
+      mod = owner.modules_hash[name] || owner.add_module(RDoc::NormalModule, name)
+    end
+
+    mod.store = @store
+    mod.line = start_line
+    record_location(mod)
+    handle_modifier_directive(mod, start_line)
+    handle_modifier_directive(mod, end_line)
+    mod.add_comment(comment, @top_level) if comment
+    mod
+  end
+
+  class RDocVisitor < Prism::Visitor # :nodoc:
+    DSL = {
```

Comment:
Can we make these `visit_call` methods more easily distinguishable from the visitor's external interface? I think adding `_` as prefix + moving them to the private section should be enough.

Reply:
I added `_` 👍 
Visibility is still public because the code below is calling method through `v.` so we can't change the visibility to private.
```ruby
-> (v, call_node) { v._visit_call_attr_reader_writer_accessor(call_node, 'R') }
```
Of course we can use instance_eval or instance_exec to remove `v.` part, but I think not using `instance_eval/exec` is more straightforward.

Reply:
Seems good and this way we can make it private!
done